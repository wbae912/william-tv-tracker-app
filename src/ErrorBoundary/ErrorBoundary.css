.error-boundary {
  text-align: center;
  color: white;
  font-family: 'Montserrat', sans-serif;
  padding-top: 100px;
  margin: 0;
}

.error-boundary-h1 {
  font-weight: bold;
  font-size: 20px;
}

.error-boundary-p {
  text-align: center;
  color: white;
  font-family: 'Montserrat', sans-serif;
  padding-top: 10px;
  font-size: 12px;
}

@media only screen and (min-width: 750px) {
  .error-boundary-h1 {
    font-size: 24px;
  }
  .error-boundary-p {
    font-size: 14px
  }
}

@media only screen and (min-width: 1300px) {
  .error-boundary-h1 {
    font-size: 30px;
  }
  .error-boundary-p {
    font-size: 16px
  }
}