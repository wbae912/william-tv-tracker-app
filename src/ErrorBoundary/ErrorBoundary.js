import React, { Component } from 'react';
import './ErrorBoundary.css';

export default class ErrorBoundary extends Component {
    constructor(props) {
        super(props)

        this.state = {
            error: false
        }
    }

    static getDerivedStateFromError(error) {
        return { error: true };
    }

    render() {
        if (this.state.error) {
            return (
                <>
                  <h2 className="error-boundary-h2">Something went wrong</h2>
                  <p className="error-boundary-p">Please refresh the page</p>
                </>
            );
        }
        return this.props.children;
    }
}