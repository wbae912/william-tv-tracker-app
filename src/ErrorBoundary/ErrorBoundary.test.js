import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import ErrorBoundary from './ErrorBoundary';
import renderer from 'react-router-dom';

describe(`ErrorBoundary component`, () => {
  const error = true;
  
  it('renders a .ErrorBoundary by default', () => {
    const wrapper = shallow(<ErrorBoundary error={error}/>)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
  // it('renders UI as expected', () => {
  //   const tree = renderer
  //     .create(<ErrorBoundary error={error}/>)
  //     .toJSON();
  //     expect(tree).toMatchSnapshot();
  // })
})