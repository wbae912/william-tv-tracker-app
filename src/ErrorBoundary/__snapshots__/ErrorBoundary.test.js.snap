// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBoundary component renders a .ErrorBoundary by default 1`] = `""`;
